### ISSUE TITLE
Cleverer `serve` action

### ISSUE THREAD

**Issue by davorg** [davorg/app-aphra#3]
Currently, the `serve` action, runs `build` and then serves a static site. Can we monitor the input directories and rebuild the site if anything changes?

**Comment by simbabque** [comment]
I will have a look at this for the Pull Request Club. Do you mind adding extra dependencies?

**Comment by davorg** [comment]
Not at all. I'd prefer them to be optional, if possible. Perhaps it could revert to the current behaviour if they aren't available.

**Comment by simbabque** [comment]
You're saying that `serve` calls `build`, but I don't see that. It just gets App::HTTPThis to serve the directory.

**Comment by davorg** [comment]
It's possible I'm describing what *should* happen rather than what *does* happen :-/

Or, maybe, you've found a bug.

**Comment by simbabque** [comment]
Do you have a demo system for me to work with?

**Comment by davorg** [comment]
Yeah. https://github.com/davorg/whonews.tv is a web site that's built using Aphra.

**Comment by simbabque** [comment]
Thanks. I have looked at  Filesys::Notify::Simple, which is what Plack::Runner/Plack::Loader uses to do reloads. I would prefer to actually use the Plack::Loader::Restarter directly instead of hacking forks into this thing, but App::HTTPThis does not support shoving in extra Plack bits as it brings its own runner that cannot easily be changed. I was thinking of patching a reload feature in, but @melo has removed the App::HTTPThis repository from his github account. I've contacted him to see what's going on there.
